jlibra-core/src/main/java/dev/jlibra/admissioncontrol/GrpcMapper.java

```diff
@@ -31,45 +18,57 @@
 import types.Transaction.RawTransaction;
 import types.Transaction.SignedTransaction;
 import types.Transaction.TransactionArgument;
-import types.Transaction.TransactionArgument.ArgType;
+
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
 
 public class GrpcMapper {
-    private static Map<Type, ArgType> jlibraArgumentTypeToGrpcArgumentType;
 
-    static {
-        jlibraArgumentTypeToGrpcArgumentType = new HashMap<>();
-        jlibraArgumentTypeToGrpcArgumentType.put(Type.ADDRESS, ArgType.ADDRESS);
-        jlibraArgumentTypeToGrpcArgumentType.put(Type.U64, ArgType.U64);
+    public static SubmitTransactionRequest toSubmitTransactionRequest(ExtendedPrivKey fromAccount, Transaction transaction, RawTransactionSigner signer) {
```

Comment:
Now we have these separate transaction creation methods because there are two different signing methods in use. One in LibraHelper and one in the ExtendedPrivateKey, but could they use the same signing? Then the transaction creation could be simpler.
If you agree, we can also make a new issue about this as it is not in the scope of the integration test.

Reply:
Unfortunately. The keys are quite different for both methods and I don't really know how to convert between them. If we find a way then we could have one signing.